perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -1198,6 +1174,7 @@ public static synchronized void refreshAuthz(PerunSession sess) throws InternalE
 		log.debug("Refreshing authz roles for session {}.", sess);
 
 		sess.getPerunPrincipal().setRoles(AuthzResolverBlImpl.getRoles(sess));
+
 		prepareServiceRoles(sess);
```

Comment:
Should be before "setting roles" to session, since when account has service role, we do not load the rest, like in ``init(sess)`` method.